### DISCUSSION TITLE
FastDDS key field in statistics topic

### DISCUSSION THREAD

**Discussion by Adjk-Luis** [eProsima/Fast-DDS#5151] [General]
I add some own custom statistics topics and configure it as fastdds builtin statistics topics, but i find that if the "        m_isGetKeyDefined" is set to false, i will not receivce the topic samples in other app which subscriber to this topic, and if set true, it is ok. Then i try to do this test on fastdds builtin statistics topics, e.g. NETWORK_LATENCY_TOPIC, it behavies as the same. And later, i do the same test using  a common example like app generated by fasddsgen, it is no probelm whether the m_isGetKeyDefined is true or false.  why this happens?

**Comment by Mario-DL** [comment]
Hi @BlueBlueSkyHaha,

Could you please provide more insight in which statistic builtin topics are you testing and have `m_isGetKeyDefined` to false ?. 
Also, ensure that you are using the [recommended qos for statistics](https://fast-dds.docs.eprosima.com/en/latest/fastdds/statistics/dds_layer/qos.html)

**Reply by Adjk-Luis** [reply]
Thanks for responce. I tested NETWORK_LATENCY_TOPIC, it is a builtin statistics topic. If i just create a app to subscribe this topic and do nothing else, i can receive the statistics sample data, but if i modify   "typesPubSubTypes.cxx" file as follow, then it did not works. 

<img width="705" alt="image" src="https://github.com/user-attachments/assets/adf46f83-a3b2-46cf-8777-ab1afc7f0f0f">

**Reply by Adjk-Luis** [reply]
sorry, it is not clear, what i modified is : m_isGetKeyDefined = true;  -> m_isGetKeyDefined = false;

**Reply by Adjk-Luis** [reply]
it seems that statistics topics must set "m_isGetKeyDefined" to true, but other user topics can set or not, I do not know if this is the truth.
